### PR TITLE
don't use PathResolver directly

### LINKED ISSUE #327: AppEngineSdkClasspathContainer should not use Pathresolver directly
private PathResolver pathResolver = PathResolver.INSTANCE;

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis @akerekes 

close #327

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java [hunk 1]
```diff
@@ -18,7 +17,7 @@
   private static final String SDK_JAR = "platform/google_appengine/google/appengine/tools/java/lib/appengine-tools-api.jar";
   public static final String CONTAINER_ID = "AppEngineSDK";
 
-  private PathResolver pathResolver = PathResolver.INSTANCE;
+  private File cloudSdkLocation = new CloudSdkProvider().getCloudSdkLocation();
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java]: So it looks fine to me.  I wonder if we should instead hold onto the `CloudSdk` instance and ask that instance for the SDK jar(s).

**Comment by briandealwis** [comment]
lgtm

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java [hunk 2]
```diff
@@ -36,11 +36,16 @@
   private final IPreferenceStore preferences;
 
    public CloudSdkProvider(IPreferenceStore preferences) {
+     // todo drop the null check and use the no args constructor instead
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java]: Can you file an issue for this? Or would it be too big of a change fixing it in this PR?

Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java]: Maybe the no-arg should use `this(PreferenceInitializer.getPreferenceStore())`

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java]: Is it likely to change while the program is running?

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java (hunk 2 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java]: done

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java [hunk 3]
```diff
@@ -44,7 +44,7 @@ public CloudSdkProvider(IPreferenceStore preferences) {
    }
    
    public CloudSdkProvider() {
-     this.preferences = PreferenceInitializer.getPreferenceStore();
+     this(PreferenceInitializer.getPreferenceStore());
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java]: Is the todo above still necessary?

**Comment by elharo** [comment]
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.cloud.tools.eclipse.appengine.newproject.CodeTemplatesTest
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated

Haven't seen this before, but I suspect this is flakiness in the test infrastructure since the pr build passed.

File: plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java (hunk 3 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java]: yes

**Comment by briandealwis** [comment]
lgtm

**Comment by akerekes** [comment]
LGTM

**Comment by elharo** [comment]
OK, passed on the manual rerun so mergin

**Pull request merged**